module/VuFind/src/VuFind/ILS/Driver/Folio.php

```diff
@@ -849,6 +855,23 @@ public function getHolding($bibId, array $patron = null, array $options = [])
                 }
                 $nextBatch[] = $nextItem;
             }
+
+            // If there are no item records at this location, we're going to create a fake one,
+            // fill it with data from the FOLIO holdings record, and make it not appear in
+            // the full record display using InvisibleAvailabilityStatus()
+            if ($number == 0) {
+                $nextBatch[] = [
+                    'callnumber' => $holdingDetails['holdingCallNumber'],
+                    'callnumber_prefix' => $holdingDetails['holdingCallNumberPrefix'],
+                    'holdings_notes' => $holdingDetails['holdingsStatements'],
+                    'indexes' => $holdingDetails['holdingsIndexes'],
+                    'supplements' => $holdingDetails['holdingsSupplements'],
+                    'location' => $holdingDetails['holdingLocationName'],
+                    'id' => $bibId,
```

Comment:
This assigning of the `'id' => $bibId` is probably the one part I'm not sure about. I found that it was necessary to do this for the AJAX/GetItemStatus::handleRequest() to work at [line 545](https://github.com/vufind-org/vufind/blob/99dec4025b7e3fe3b418e2365bc6b2812dbe3604/module/VuFind/src/VuFind/AjaxHandler/GetItemStatuses.php#L545) where it uses this `id` value to match up against the bib record identifier; that is important because later at line 549 we declare that the record has been "found". But this is technically the holdings record identifier?

Reply:
VuFind doesn't use holdings IDs, it only uses bib IDs and item IDs -- the ILS driver spec flattens everything out. So I think this is probably okay.